Love 0.7 broken on master and 19.03

**Describe the bug**

Love 0.7 doesn't build at the moment.

```
nix build -f channel:nixos-19.03 love_0_7
builder for '/nix/store/givchn1sx30qh0swhga4rsprqnzg6w6v-love-0.7.2.drv' failed with exit code 1; last 10 log lines:
  checking dynamic linker characteristics... GNU/Linux ld.so
  checking how to hardcode library paths into programs... immediate
  checking for library containing sqrt... -lm
  checking for library containing SDL_Init... -lSDL
  checking for library containing glLoadIdentity... -lGL
  checking for library containing alSourcePlay... -lopenal
  checking for library containing lua_pcall... -llua
  checking for library containing ilInit... -lIL
  checking for library containing mng_initialize... no
  configure: error: DevIL needs MNG
[0 built (1 failed)]
error: build of '/nix/store/givchn1sx30qh0swhga4rsprqnzg6w6v-love-0.7.2.drv' failed
````

After adding libmng it fails during compiling.

**To Reproduce**

`nix build -f channel:nixos-19.03 love_0_7`

**Expected behavior**
A clear and concise description of what you expected to happen.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 4.20.0, NixOS, 19.03.173214.4cc5592fe2d (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - channels(root): `"home-manager-19.03, nixos-19.03.173202.31d476b8797, nixos-unstable-19.09pre186857.239fffc90d7, nixpkgs-19.03"`
 - channels(addy): `"nixos-19.03.173214.4cc5592fe2d, nixos-unstable-19.09pre186857.239fffc90d7, nixpkgs-19.03.173214.4cc5592fe2d"`
 - nixpkgs: `/home/addy/.nix-defexpr/channels/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: love_0_7 rimshot
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/development/interpreters/love/0.7-gl-prototypes.patch
+++ b/pkgs/development/interpreters/love/0.7-gl-prototypes.patch
@@ -1,0 +1,21 @@
+diff --unified --recursive --text love-HEAD.orig/src/modules/graphics/opengl/Framebuffer.cpp love-HEAD.new/src/modules/graphics/opengl/Framebuffer.cpp
+--- love-HEAD.orig/src/modules/graphics/opengl/Framebuffer.cpp	2019-03-14 12:46:55.032982224 -0400
++++ love-HEAD.new/src/modules/graphics/opengl/Framebuffer.cpp	2019-03-14 12:47:22.356175299 -0400
+@@ -1,3 +1,5 @@
++#define GL_GLEXT_PROTOTYPES
++
+ #include "Framebuffer.h"
+ #include <common/Matrix.h>
+ 
+diff --unified --recursive --text love-HEAD.orig/src/modules/graphics/opengl/SpriteBatch.cpp love-HEAD.new/src/modules/graphics/opengl/SpriteBatch.cpp
+--- love-HEAD.orig/src/modules/graphics/opengl/SpriteBatch.cpp	2019-03-14 12:46:55.032982224 -0400
++++ love-HEAD.new/src/modules/graphics/opengl/SpriteBatch.cpp	2019-03-14 12:47:33.346119890 -0400
+@@ -18,6 +18,8 @@
+ * 3. This notice may not be removed or altered from any source distribution.
+ **/
+ 
++#define GL_GLEXT_PROTOTYPES
++
+ #include "SpriteBatch.h"
+ 
+ // STD

--- a/pkgs/development/interpreters/love/0.7.nix
+++ b/pkgs/development/interpreters/love/0.7.nix
@@ -2,6 +2,7 @@
 , SDL, libGLU_combined, openal, lua
 , libdevil, freetype, physfs
 , libmodplug, mpg123, libvorbis, libogg
+, libmng
 }:
 
 stdenv.mkDerivation rec {
@@ -11,10 +12,13 @@ stdenv.mkDerivation rec {
     sha256 = "0s7jywkvydlshlgy11ilzngrnybmq5xlgzp2v2dhlffwrfqdqym5";
   };
 
+  # see discussion on arch linux user repository (https://aur.archlinux.org/packages/love07/?setlang=cs#comment-684696)
+  patches = [ ./0.7-gl-prototypes.patch ];
+
   nativeBuildInputs = [ pkgconfig ];
   buildInputs = [
     SDL libGLU_combined openal lua
-    libdevil freetype physfs libmodplug mpg123 libvorbis libogg
+    libdevil freetype physfs libmodplug mpg123 libvorbis libogg libmng
   ];
 
   preConfigure = ''

